discovery.md

```diff
@@ -0,0 +1,436 @@
+# CloudSubscriptions: Discovery - Version 0.1-rc01
+
+## Abstract
+
+CloudSubscriptions Discovery API is a vendor-neutral API specification for
+determining what events an Event Producer or Aggregator / Broker
+has available, as well as how to subscribe to those events.
+
+## Status of this document
+
+This is a working draft. Breaking changes could be made in the next minor
+version.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Notations and Terminology](#notations-and-terminology)
+- [API Specification](#api-specification)
+- [Protocol Bindings](#protocol-bindings)
+- [Privacy & Security](#privacy-and-security)
+
+## Overview
+
+The goal of the CloudDiscovery API is to enable connection between consumers of
+events and producers / aggregators / brokers of events and to facilitate the
+creation of CloudSubscriptions.
+
+Discovery must allow for an event producer or intermediary component to
+advertise the event types that are available, provide the necessary information
+to consume that event (schema / delivery protocol options), and the necessary
+information to create a subscription. The output of the API must be such that
+tooling can be built where all possible event producers and types aren’t known
+in advance.
+
+There are several discovery use cases to consider from the viewpoint of event
+consumers.
+
+1. What sources are available, and what event types do they produce, and
```

Comment:
It seems like "sources" and the CE-source attribute are different in this sentence. Is source == producer? Will help to provide links to producer and source definition if they are present somewhere else.